### PR TITLE
Added types for npm guid package

### PR DESCRIPTION
**Pull request by maroy1986**
I'm opening this PR to add missing definition for the npm package `guid` (https://github.com/dandean/guid).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/guid/index.d.ts
Comment (sheetalkamat) [inline types/guid/index.d.ts]: why do you have namepsaces for the functions? they are not needed ?

Comment (maroy1986) [reply types/guid/index.d.ts]: You have a good point, those are not required after verification, they just get generated by `dts-gen`. I'll get them removed.

**Review by plantain-00 — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by third party. PR appears ready to merge pending review by a maintainer.

**Pull request merged**
